From: Snild Dolkow <snild@sony.com>
Date: Thu, 28 Sep 2023 13:06:39 +0200
Subject: tests: Replace invalid entity expansion in
 test_alloc_nested_entities

%pe2; would ultimately expand to a plain "ABCDEF...", which is not
valid in this context. This was not normally hit, since the test would
get its expected XML_ERROR_NO_MEMORY before expanding this far.

With g_chunkSize=0 and EXPAT_CONTEXT_BYTES=OFF, the number of allocs
required to reach that point becomes *just* low enough to reach the
final expansion, making the test fail with a very unexpected syntax
error.

Nesting %pe2; in another entity declaration avoids the problem.

Origin: https://github.com/libexpat/libexpat/commit/7b0e27a6981014313a9f30486a0b4d2e0a3ebde3
---
 expat/tests/runtests.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/expat/tests/runtests.c b/expat/tests/runtests.c
index 554d485..2da59a7 100644
--- a/expat/tests/runtests.c
+++ b/expat/tests/runtests.c
@@ -9958,7 +9958,7 @@ START_TEST(test_alloc_nested_entities) {
          "ABCDEFGHIJKLMNOPABCDEFGHIJKLMNOPABCDEFGHIJKLMNOPABCDEFGHIJKLMNOP"
          "'>\n"
          "<!ENTITY % pe2 '%pe1;'>\n"
-         "%pe2;",
+         "<!ENTITY % pe3 '%pe2;'>",
          "Memory Fail not faulted", NULL, XML_ERROR_NO_MEMORY};
 
   /* Causes an allocation error in a nested storeEntityValue() */
